common/ts/helpers/request.ts

```diff
@@ -7,59 +7,33 @@ interface RequestData {
   [x: string]: any;
 }
 
-function get(endpoint: Endpoint, path: string, isJson: boolean, query?: RequestData): Promise<any> {
+export function get(endpoint: Endpoint, path: string, query?: RequestData): Promise<any> {
   tl.debug(`[SQ] API GET: '${path}' with query "${JSON.stringify(query)}"`);
-  return new Promise((resolve, reject) => {
-    const options: request.CoreOptions = {
-      auth: endpoint.auth,
-    };
-    if (query) {
-      options.qs = query;
-      options.useQuerystring = true;
-    }
-    request.get(
-      {
-        method: "GET",
-        baseUrl: endpoint.url,
-        uri: path,
-        json: isJson,
-        ...options,
-      },
-      (error, response, body) => {
-        if (error) {
-          return logAndReject(
-            reject,
-            `[SQ] API GET '${path}' failed, error was: ${JSON.stringify(error)}`
-          );
-        }
-        tl.debug(
-          `Response: ${response.statusCode} Body: "${isString(body) ? body : JSON.stringify(body)}"`
-        );
-        if (response.statusCode < 200 || response.statusCode >= 300) {
-          return logAndReject(
-            reject,
-            `[SQ] API GET '${path}' failed, status code was: ${response.statusCode}`
-          );
-        }
-        return resolve(body || (isJson ? {} : ""));
-      }
-    );
-  });
-}
 
-function isString(x) {
-  return Object.prototype.toString.call(x) === "[object String]";
-}
+  return axios({
+    url: path,
+    method: "get",
+    baseURL: endpoint.url,
+    auth: {
+      username: endpoint.auth.user,
+      password: endpoint.auth.pass,
+    },
+    params: query,
+  })
+    .then((response) => {
+      tl.debug(`Response: ${response.status} Body: "${response.data}"`);
 
-export function getJSON(endpoint: Endpoint, path: string, query?: RequestData): Promise<any> {
-  return get(endpoint, path, true, query);
+      return response.data;
+    })
+    .catch((error) => {
+      if (error.response) {
+        tl.debug(`[SQ] API GET '${path}' failed, status code was: ${error.response.status}`);
+      } else {
+        tl.debug(`[SQ] API GET '${path}' failed, error is ${error.message}`);
+      }
```

Comment:
You dropped `logAndReject` fn which makes sense to me ( I don't see any benefit of it), but I think we should throw an error here too.  Otherwise any task will continue to work without the response we should have received, if I understand correctly.

Reply:
Yes I was a bit undecided about that, because I saw the error was not handled and the reject was giving the same information as the log. But I'm fine either way, happy to throw something to be safe